lib/nerves_runtime/heart.ex

```diff
@@ -0,0 +1,131 @@
+  Nerves Heart's current status
+
+  * `:program_name` - `"nerves_heart"`
+  * `:program_version` - Nerves heart's version number
+  * `:identity` - The hardware watchdog that's being used
+  * `:firmware_version` - An integer that represents the hardware watchdog's firmware revision
+  * `:options` - Hardware watchdog options as reported by Linux
+  * `:time_left` - How many seconds are left before the hardware watchdog triggers a reboot
+  * `:pre_timeout` - How many seconds before the watchdog expires that Linux
+    will receive a pre-timeout notification * `:timeout` - The hardware watchdog
+    timeout. This is only changeable in the Linux configuration
+  * `:last_boot` - What caused the most recent boot. Whether this is reliable
+    depends on the watchdog.
+  * `:heartbeat_timeout` - Erlang's heartbeat timeout setting. Note that the
+    hardware watchdog timeout supersedes this since it reboots.
+  """
+  @type info() :: %{
+          program_name: String.t(),
+          program_version: Version.t(),
+          identity: String.t(),
+          firmware_version: non_neg_integer(),
+          options: non_neg_integer() | [atom()],
+          time_left: non_neg_integer(),
+          pre_timeout: non_neg_integer(),
+          timeout: non_neg_integer(),
+          last_boot: :power_on | :watchdog,
+          heartbeat_timeout: non_neg_integer()
+        }
+
+  @doc """
+  Return whether Nerves heart is running
+
+  If you're using a Nerves device, this always returns `true` except possibly
+  when porting Nerves to new hardware. It is a quick sanity check.
+  """
+  def running_nerves_heart?() do
+    case status() do
+      {:ok, %{program_name: "nerves_heart"}} -> true
+      _ -> false
+    end
+  end
+
+  @doc """
+  Return the current Nerves Heart status
+
+  Errors are returned when not running Nerves Heart
+  """
+  @spec status() :: {:ok, info()} | :error
+  def status() do
+    {:ok, cmd} = :heart.get_cmd()
+
+    parse_cmd(cmd)
+  rescue
+    ArgumentError ->
+      # When heart isn't running, an ArgumentError is raised
+      :error
+  end
+
+  @doc """
+  Raising version of status/0
+  """
+  @spec status!() :: info()
+  def status!() do
+    {:ok, results} = status()
+    results
+  end
+
+  @doc false
+  @spec parse_cmd(list()) :: {:ok, info()} | :error
+  def parse_cmd([]), do: :error
+
+  def parse_cmd(cmd) when is_list(cmd) do
+    result =
+      cmd
+      |> to_string()
+      |> String.split("\n")
+      |> Enum.map(&String.split(&1, "=", parts: 2))
+      |> Enum.map(&parse_attribute/1)
+      |> Enum.reject(&is_nil/1)
+      |> Map.new()
+
+    {:ok, result}
+  rescue
+    _ -> :error
+  end
+
+  defp parse_attribute(["program_name", str]), do: {:program_name, str}
+  defp parse_attribute(["program_version", str]), do: {:program_version, Version.parse!(str)}
+  defp parse_attribute(["identity", str]), do: {:identity, str}
+  defp parse_attribute(["firmware_version", str]), do: {:firmware_version, parse_int(str)}
```

Comment:
```suggestion
  defp parse_attribute(["firmware_version", str]), do: {:firmware_version, String.to_integer(str)}
```
Is there a particular reason to avoid standard lib here for the same effect? (You can also do `String.to_integer(str, 16)` below)

Reply:
Thanks. I was going quickly and forgot about `String.to_integer/2`.